app_package/src/watch.ts

```diff
@@ -0,0 +1,325 @@
+import { AbstractMesh, AnimationGroup, Bone, ISceneLoaderAsyncResult, Matrix, Observable, PBRMaterial, Scene, SceneLoader, Texture, TmpVectors, Tools, TransformNode, Vector2, Vector3 } from "@babylonjs/core";
+import { AdvancedDynamicTexture, TextBlock } from "@babylonjs/gui";
+import { IVaporwearExperienceParams } from "./iVaporwearExperienceParams";
+
+export enum WatchState {
+    Overall,
+    Clasp,
+    Face,
+    Levitate,
+    Configure
+}
+
+class WatchStateHelpers {
+    public static IsUpState(state: WatchState): boolean {
+        switch (state) {
+            case WatchState.Overall:
+            case WatchState.Clasp:
+            case WatchState.Configure:
+                return true;
+            default:
+                return false;
+        }
+    }
+}
+
+export class HotspotState {
+    constructor () {
+        this.position = new Vector2();
+        this.isVisible = false;
+    }
+
+    position: Vector2;
+    isVisible: boolean;
+}
+
+export class Watch extends TransformNode {
+    private _animationWatchSpinUp: AnimationGroup;
+    private _animationWatchSpinDown: AnimationGroup;
+    private _animationOrbitOverall: AnimationGroup;
+    private _animationOrbitClasp: AnimationGroup;
+    private _animationOrbitFace: AnimationGroup;
+    private _animationOrbitLevitate: AnimationGroup;
+    private _animationHotspot0Visibility: AnimationGroup;
+    private _animationHotspot1Visibility: AnimationGroup;
+
+    private _bodyBone: Bone;
+    private _rootMesh: AbstractMesh;
+
+    private _cameraParentOverall: TransformNode;
+    private _cameraParentClasp: TransformNode;
+    private _cameraParentFace: TransformNode;
+    private _cameraParentLevitate: TransformNode;
+
+    private _hotspot0: TransformNode;
+    private _hotspot1: TransformNode;
+    private _hotspot2: TransformNode;
+    private _hotspot0Visibility: TransformNode;
+    private _hotspot1Visibility: TransformNode;
+    private _hotspot2Visibility: TransformNode;
+
+    public hotspot0State: HotspotState;
+    public hotspot1State: HotspotState;
+    public hotspot2State: HotspotState;
+    public onHotspotsUpdatedObservable: Observable<void>;
+
+    private _state: WatchState;
+
+    public get cameraParentOverall(): TransformNode {
+        return this._cameraParentOverall;
+    }
+
+    public get cameraParentClasp(): TransformNode {
+        return this._cameraParentClasp;
+    }
+
+    public get cameraParentFace(): TransformNode {
+        return this._cameraParentFace;
+    }
+
+    public get cameraParentLevitate(): TransformNode {
+        return this._cameraParentLevitate;
+    }
+
+    private constructor (scene: Scene, importWatchResult: ISceneLoaderAsyncResult) {
```

Comment:
nit: maybe you should format all the files with a formatter
```suggestion
    private constructor(scene: Scene, importWatchResult: ISceneLoaderAsyncResult) {
```